docs/bir-spec/src/test/java/org/ballerinalang/birspec/BIRTestUtils.java

```diff
@@ -194,14 +196,30 @@ private static int generateExpectedScopeEntries(List<BIRNode.BIRBasicBlock> bbLi
                 }
 
                 visitedScopes.add(instructionScope);
-                boolean hasParent = instructionScope.parent != null;
+                hasParent = instructionScope.parent != null;
 
-                ExpectedScopeEntry expectedScopeEntry = new ExpectedScopeEntry(instructionScope.id,
+                expectedScopeEntry = new ExpectedScopeEntry(instructionScope.id,
                         instructionOffset,  hasParent ? 1 : 0, hasParent ? instructionScope.parent.id : null);
                 scopes.put(instructionScope.id, expectedScopeEntry);
                 putParentScopesAsWell(scopes, instructionScope.parent, instructionOffset);
 
             }
+            BIRTerminator terminator = bb.terminator;
+            BirScope terminatorScope = terminator.scope;
+
+            if (terminatorScope != null) {
+
+                if (visitedScopes.contains(terminatorScope)) {
+                    continue;
+                }
+
+                visitedScopes.add(terminatorScope);
```

Comment:
Why we need this change?

Reply:
Previous BirSpec test only considered scopes of non-terminator instructions and scopes for terminators also generated with this change. In order to incorporate the new changes with the existing test, these changes are needed.